llvmDebugInfoC/src/scripts/konan_lldb.py

```diff
@@ -92,19 +103,20 @@ def kotlin_object_type_summary(lldb_val, internal_dict):
 
 
 def select_provider(lldb_val):
-    return __FACTORY['string'](lldb_val) if is_string(lldb_val) else __FACTORY['array'](lldb_val) if is_array(
-        lldb_val) else __FACTORY['object'](lldb_val)
-
+    soa = is_string_or_array(lldb_val)
+    return __FACTORY['string'](lldb_val) if soa == 1 else __FACTORY['array'](lldb_val) if soa == 2 \
+        else __FACTORY['object'](lldb_val)
 
 class KonanHelperProvider(lldb.SBSyntheticValueProvider):
-    def __init__(self, valobj):
+    def __init__(self, valobj, amString):
```

Comment:
Hacky. String should probably be its own class. The purpose here is to avoid the extra 'Konan_DebugGetFieldCount' call.